### PR TITLE
use triplet cycle consistency in scene optimizer

### PR DESCRIPTION
**Pull request by johnwlambert**
Previously the module was added, but now incorporating this into the main workflow.

### CODE REVIEW

File: gtsfm/scene_optimizer.py [hunk 1]
```diff
@@ -174,6 +175,14 @@ def create_computation_graph(
         keypoints_graph_list = dask.delayed(lambda x, y: (x, y))(keypoints_graph_list, auxiliary_graph_list)[0]
         auxiliary_graph_list = []
 
+        # ensure cycle consistency in triplets
+        cycle_consistent_graph = dask.delayed(cycle_utils.filter_to_cycle_consistent_edges)(
+            i2Ri1_graph_dict, i2Ui1_graph_dict, two_view_reports_dict
+        )
+
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:182]: nit: this empty line is unnecessary. The next two lines are very much related to the previous.

File: gtsfm/scene_optimizer.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 from dask.delayed import Delayed
 
+import gtsfm.averaging.rotation.cycle_consistency as cycle_utils
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:17]: This is not a util, just import it as cycle_consistency?

**Review by akshay-krishnan — CHANGES_REQUESTED** [review]

File: gtsfm/scene_optimizer.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py]: good point, updated now

File: gtsfm/scene_optimizer.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py]: good point, updated now

**Review by akshay-krishnan — APPROVED** [review]

**Comment by ayushbaid** [comment]
Let's discuss the metrics for this branch.

**Comment by johnwlambert** [comment]
@ayushbaid  Averaging Metrics:

Metric name | Value Before | Value After
-- | -- | --
mean_rotation_averaging_angle_deg | 10.51 | 7.95
mean_translation_averaging_distance | 1.30 | 0.87 
mean_translation_angle_deg | 45.08 | 31.50

**Comment by johnwlambert** [comment]
Data Association Metrics

Metric name | Value Before | Value After
-- | -- | --
mean_2D_track_lengths | 2.81513 |  2.94767
accepted_tracks_ratio | 0.651628 | 0.670509
track_cheirality_failure_ratio | 0.272727 | 0.252915
num_accepted_tracks | 3842 | 5061
mean_3d_tracks_length | 2.56273 | 2.65679
mean_accepted_track_avg_errors_px | 26.0867 | 22.2262
mean_rejected_track_avg_errors_px | 95.5614 | 101.426

**Pull request merged**